genesyscloud/integration_credential/resource_genesyscloud_integration_credential.go

```diff
@@ -41,16 +43,27 @@ utils function in the package.  This will keep the code manageable and easy to w
 // getAllCredentials retrieves all of the integration credentials via Terraform in the Genesys Cloud and is used for the exporter
 func getAllCredentials(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
 	resources := make(resourceExporter.ResourceIDMetaMap)
-	ip := getIntegrationCredsProxy(clientConfig)
+	integrationCredsProxy := getIntegrationCredsProxy(clientConfig)
+	integrationsProxy := integration.GetIntegrationsProxy(clientConfig)
 
-	credentials, err := ip.getAllIntegrationCreds(ctx)
+	credentials, err := integrationCredsProxy.getAllIntegrationCreds(ctx)
 	if err != nil {
 		return nil, diag.Errorf("Failed to get all credentials: %v", err)
 	}
 
 	for _, cred := range *credentials {
 		log.Printf("Dealing with credential id : %s", *cred.Id)
 		if cred.Name != nil { // Credential is possible to have no name
+			// Verify that the integration entity itself exist before exporting the integration credentials associated to it: DEVTOOLING-282
+			integrationId := strings.Split(*cred.Name, "Integration-")[1]
+			_, resp, err := integrationsProxy.GetIntegrationById(ctx, integrationId)
+			if err != nil {
+				if gcloud.IsStatus404(resp) {
```

Comment:
We should put a log message here indicating that we hit this part of the code

Reply:
Agreed, done!